src/catkin_pkg/condition.py

```diff
@@ -58,6 +58,20 @@ def _evaluate(parse_results, context):
         return parse_results
 
     # recursion
+    if len(parse_results) != 3:
```

Comment:
The following code only works for specific `len(oarse_results)`, e.g. `2` wouldn't work. Please add an assert to only handle those explicitly.

Reply:
It it really the case that `parse_results` can have any number: `4`, `5`, `6`, ...?

(If yes, the following `if` can be removed since the `assert` ensures it is always `True`.)

Reply:
`len(parse_results)` is supposed to be a odd number, So, I have put that condition in `if`.